File: day_06/main.R
library(readr)
library(magrittr)

find_index <- function(input, n) {
  for (i in n:(length(input) - n)) {
    if ((input[(i - n):(i - 1)] %>% unique() %>% length()) == n) {
      return(i - 1)
    }
  }
}

input_data <- read_file_raw("input.txt")

cat("Part one: ", find_index(input_data, 4))
cat("Part two: ", find_index(input_data, 14))

File: day_06/main.py
def index_after_n_unique_chars(n: int, input_str: str) -> int:
    for i in range(n, len(input_str)-n):
        if len(set(input_str[i-n:i])) == n:
            return i


def main() -> None:
    with open('input.txt', 'r') as f:
        input_data = f.read()

    print('Part one: ', index_after_n_unique_chars(4, input_data))
    print('Part two: ', index_after_n_unique_chars(14, input_data))


if __name__ == '__main__':
    main()
